### PR TITLE
Update `ChoiceList` to use Layout primitives

### LINKED ISSUE #7354: [Layout foundations] Rebuild `ChoiceList` component with new layout primitives
https://github.com/Shopify/polaris/issues/7354#tasklist-block-8e94a794-6490-43df-9e31-9fce96a3fa12

### PR DESCRIPTION
**Pull request by aveline**
### WHY are these changes introduced?

Resolves #7354

### WHAT is this pull request doing?

- Refactored `ChoiceList` to use primitive Layout components
- Added support for `legend` element to `Box`
- Add support for `fieldset` element to `AlphaStack`

### How to 🎩

There is only one small visual change for `ChoiceList`s with Help text on small screens only. There is `space-1` additional spacing between items which is `margin` on Help Text that long longer overlaps into the space between `Choices` now that the space is set with `gap` in `AlphaStack`. Previously it was set with `margin` so they overlapped.

Screenshot 2022-11-17 at 8 48 15 AM

Otherwise there should be no visual changes.

### CODE REVIEW

File: .changeset/cold-keys-raise.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@shopify/polaris': minor
+---
+
+- Refactored `ChoiceList` to use primitive Layout components
+- Added support for `legend` element to `Box`
+- Add support for `fieldset` element to `AlphaStack`
```
Comment (laurkim) [inline .changeset/cold-keys-raise.md]:
```suggestion
- Added support for `fieldset` element to `AlphaStack`
```

File: polaris-react/src/components/ChoiceList/ChoiceList.scss [hunk 2]
```diff
@@ -1,50 +1,5 @@
...
-  padding: 0;
-  list-style: none;
-}
-
 .ChoiceChildren {
-  @media #{$p-breakpoints-md-down} {
-    margin-top: var(--p-space-4);
-  }
-  margin-bottom: var(--p-space-2);
   padding-left: calc(var(--p-space-2) + var(--p-choice-size));
```
Comment (laurkim) [inline polaris-react/src/components/ChoiceList/ChoiceList.scss:4]:
```suggestion
  padding-left: calc(var(--p-space-2) + var(--p-space-5));
```

Looks like `--p-choice-size` is a `20px` legacy token that we already have in spacing tokens.

**Review by laurkim — APPROVED** [review]
Nice! 💯

**Review by kyledurand — APPROVED** [review]
👏

File: polaris-react/src/components/ChoiceList/ChoiceList.scss (hunk 2 above)
Comment (aveline) [reply polaris-react/src/components/ChoiceList/ChoiceList.scss:4]: https://github.com/Shopify/polaris/blob/main/polaris-react/src/components/Choice/Choice.scss#L47

`--p-choice-size` changes based on breakpoint so can't be replaced with the spacing token.

Comment (laurkim) [reply polaris-react/src/components/ChoiceList/ChoiceList.scss:4]: Ahh, missed that line thank you for the context!

**Review by sarahill — COMMENTED** [review]
@aveline this looks ok. I would prefer to keep spacing between `16px` instead of the `20px` due to the `space-1` but I understand why that's there. A potential alternative is to make the spacing on the help text responsive too but it may not be worth adding that complexity for this. Thoughts?

**Comment by aveline** [comment]
@sarahill the help text is part of `Choice`, not `ChoiceList` so it's hard to make the change only in `ChoiceList` and not create some unexpected changes when `Choice` is used standalone or in different contexts.

But you did give me an idea, maybe I can add a responsive `Bleed` to handle that spacing in `ChoiceList` I will give it a try. It's probably not a great pattern but good exploration in terms of stress testing these layout primitives.

**Comment by sarahill** [comment]
Cool I'll trust your judgement on using `Bleed`. If we would rather not I'm ok with that. Thanks for talking it out

**Review by sarahill — APPROVED** [review]

**Pull request merged**